src/context/filter.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{
+    Context, RedisModuleCommandFilter, RedisModuleCommandFilterCtx, RedisModuleString,
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_RegisterCommandFilter, RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::c_int;
+use std::str::Utf8Error;
+
+#[derive(Debug, Clone, Copy)]
+pub struct CommandFilter {
+    inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+impl CommandFilter {
+    pub fn new(inner: *mut RedisModuleCommandFilter) -> Self {
+        CommandFilter { inner }
+    }
+
+    pub fn is_null(&self) -> bool {
+        self.inner.is_null()
+    }
+}
+
+pub struct CommandFilterCtx {
+    inner: *mut RedisModuleCommandFilterCtx,
+}
+
+impl CommandFilterCtx {
```

Comment:
If we want documentation for this structure and the one above to show up in Documentation, we would need to add documentation style comments for the struct and the functions.

Example:
https://docs.rs/valkey-module/latest/valkey_module/struct.Context.html

Reply:
added documentation on both `CommandFilterCtx` and `impl Context `